File: backend/routes/apartments.py
# Apartment Routes

from fastapi import APIRouter
from datetime import datetime, timezone
import uuid
from models.schemas import ApartmentInquiryCreate
from services.database import add_to_collection

router = APIRouter(prefix="/apartment-inquiries", tags=["Apartment Inquiries"])


@router.post("")
async def submit_apartment_inquiry(inquiry: ApartmentInquiryCreate):
    """Submit an apartment partnership inquiry"""
    new_inquiry = {
        "id": str(uuid.uuid4()),
        "contactName": inquiry.contactName,
        "email": inquiry.email,
        "phone": inquiry.phone,
        "apartmentName": inquiry.apartmentName,
        "city": inquiry.city,
        "unitsCount": inquiry.unitsCount,
        "message": inquiry.message,
        "createdAt": datetime.now(timezone.utc).isoformat(),
        "status": "pending"
    }
    
    add_to_collection("apartmentInquiries", new_inquiry)
    
    return {
        "success": True,
        "message": "Thank you for your interest in partnering with ZestDo! Our team will contact you within 24 hours.",
        "inquiry": new_inquiry
    }

File: backend/routes/apartments.js
/**
 * Apartment Inquiry Routes
 */

const express = require('express');
const router = express.Router();
const { v4: uuidv4 } = require('uuid');
const { addToCollection } = require('../services/database');

// Submit apartment partnership inquiry
router.post('/', (req, res) => {
  const { contactName, email, phone, apartmentName, city, unitsCount, message } = req.body;

  // Validation
  if (!contactName || !email || !apartmentName) {
    return res.status(400).json({
      error: 'Missing required fields',
      required: ['contactName', 'email', 'apartmentName']
    });
  }

  const newInquiry = {
    id: uuidv4(),
    contactName,
    email,
    phone: phone || null,
    apartmentName,
    city: city || 'Bengaluru',
    unitsCount: unitsCount || null,
    message: message || null,
    createdAt: new Date().toISOString(),
    status: 'pending'
  };

  addToCollection('apartmentInquiries', newInquiry);

  res.status(201).json({
    success: true,
    message: 'Thank you for your interest in partnering with ZestDo! Our team will contact you within 24 hours.',
    inquiry: newInquiry
  });
});

module.exports = router;
